src/genotype.cpp

```diff
@@ -116,10 +122,14 @@ Genotype::Genotype(const std::filesystem::path& model_file){
                 });
         }
 
-        /**
-         * FIXME: After adding the graph representation of the network,
-         * FIXME: also construct the graph HERE!
-         */
+        // construct graph representation of the initial network
+        for(Connection connection : connection_genes){
+                graph[connection.in];
+        }
+
+        for (Connection connection : connection_genes){
+                graph.find(connection.in)->second.push_back(connection.out);
```

Comment:
When adding edges, we don't want to add ALL of them. We only want to add the enabled edges. So need an if statement to check. If an edge is enabled afterwards, we'll adjust both representation accordingly.